### PR TITLE
better options mgr & improve json.more equality check & remove jsonnull

### PR DESCRIPTION
**Pull request by gregsdennis**
Adds the ability to specify options for the options mgr.

Defaults to unsafe encoding for JsonSchema.

Make equality check more explicit (`.ToJsonString()` uses reflection).

Removes JsonNull because it's causing problems.

### CODE REVIEW

**Comment by gregsdennis** [comment]
One day I'll get the right base branch!

File: JsonSchema/ErrorMessages.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public static string ReplaceTokens(this string message, params (string token, ob
 		{
 			var parameter = parameters[i];
 #pragma warning disable IL2026, IL3050
-			values[i] = JsonSerializer.Serialize(parameter.value, JsonSchemaSerializerContext.OptionsManager.SerializerOptionsUnsafeRelaxedJsonEscaping);
+			values[i] = JsonSerializer.Serialize(parameter.value, JsonSchemaSerializerContext.OptionsManager.SerializerOptions);
```
Comment (jevansaks) [inline JsonSchema/ErrorMessages.cs:56]: Does this mean we don't need that unsafe version of the SerializerOptions anymore?

Comment (jevansaks) [reply JsonSchema/ErrorMessages.cs:56]: Or does it mean all of the usages of SerializerOptions are now unsafe? I'm not super familiar with this, but reading the documentation it seems like we probably shouldn't be using this option broadly.

File: JsonSchema/JsonSchemaSerializerContext.cs [hunk 2]
```diff
@@ -99,8 +101,9 @@ internal partial class JsonSchemaSerializerContext : JsonSerializerContext
 	static JsonSchemaSerializerContext()
 	{
 		OptionsManager = new TypeResolverOptionsManager(
+			new JsonSerializerOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping }
```
[the remark this answers is not in the document]
Comment (gregsdennis) [reply JsonSchema/JsonSchemaSerializerContext.cs:104]: At the end of a sequence, yes, but not at the end of a parameter list.

File: JsonSchema/ErrorMessages.cs (hunk 1 above)
Comment (gregsdennis) [reply JsonSchema/ErrorMessages.cs:56]: I didn't realize this was the only place this was being used.

**Comment by gregsdennis** [comment]
Apparently the build for JsonSchema.DataGeneration.Tests failed.  Works fine locally.  Since I didn't change anything with it, I'm okay with merging this PR.  If it continues, that's a problem for future me.

File: JsonSchema/ErrorMessages.cs [hunk 3]
```diff
@@ -20,12 +19,30 @@ namespace Json.Schema;
...
 	/// Gets or sets a culture to use for error messages.  Default is <see cref="CultureInfo.CurrentCulture"/>.
 	/// </summary>
 	public static CultureInfo? Culture { get; set; }
 
+	static ErrorMessages()
+	{
+		_serializerOptions = new JsonSerializerOptions
+		{
+#if NET8_0_OR_GREATER
+			TypeInfoResolver = JsonSchemaSerializerContext.OptionsManager.TypeInfoResolver
```
Comment (jevansaks) [inline JsonSchema/ErrorMessages.cs:34]:
```suggestion
			TypeInfoResolver = JsonSchema.TypeInfoResolver;
```

File: JsonSchema/ErrorMessages.cs [hunk 4]
```diff
@@ -20,12 +19,30 @@ namespace Json.Schema;
...
+#endif
+		};
+#if NET8_0_OR_GREATER
+		JsonSchemaSerializerContext.OptionsManager.TypeInfoResolverUpdated +=
+			(_, _) => _serializerOptions = new JsonSerializerOptions
+			{
+				TypeInfoResolver = JsonSchemaSerializerContext.OptionsManager.TypeInfoResolver
```
Comment (jevansaks) [inline JsonSchema/ErrorMessages.cs:41]:
```suggestion
				TypeInfoResolver = JsonSchema.TypeInfoResolver;
```

**Review by jevansaks — APPROVED** [review]
Approved with suggestions.

**Pull request merged**